Add community values, improve code of conduct (#1655)
open-telemetry/community: mission-vision-values.md

want to attract a diverse group of people to collaborate with us, which means
acknowledging that people come from different backgrounds and cultures. 

There might be situations where community members act in a dubious manner. If
you have seen or experienced unacceptable behavior or anything that would make
our community less welcoming, please speak up! See [our code of
conduct](https://github.com/open-telemetry/community/blob/main/code-of-conduct.md)
for more information on how to report unacceptable behavior.

While we want to encourage everyone to express themselves in their own way,
there are some behaviors that we encourage you to adopt while interacting with
other community members.

### Act on behalf of the project

It’s no secret that a good number of maintainers of the project are employed by
companies with commercial interests in OpenTelemetry, especially vendors in the
observability space. That said, we expect community members to act in the best
interests of the project. Each member’s priorities can (and should!) align with
those of their employers so that the relationship is beneficial to all parties,

> but when acting as a maintainer or contributor to the project, community members

Reviewer: Suggestion: Change sentence to --> Each member’s priorities when acting as a maintainer or contributor for the project should represent the interests of the project and its community first before their other obligations (e.g. employer).

are expected to wear the project’s hat. 

### Disclose potential conflicts of interest

> Even within the project, people might have different hats: a Collector

Reviewer: Change "might" to "may".

OpenTelemetry users should not have to choose between high-quality telemetry and
a performant application. High performance is a requirement for OpenTelemetry,
and unexpected interference effects in the host application are unacceptable.

## Community values &mdash; the principles that guide our interactions

The OpenTelemetry project aims to be a welcoming place where new and existing
members feel safe to respectfully share their opinions and disagreements. We
want to attract a diverse group of people to collaborate with us, which means

> acknowledging that people come from different backgrounds and cultures. 

Reviewer 2: This seems like a copy of what is in code-of-conduct.md. Any reason we need this in 2 places?

Author: I think it's a good introduction to what comes next, serving both purposes well (mission/vision/values and CoC). As I see no harm in having it at both places, I'll leave it there for now.

There might be situations where community members act in a dubious manner. If
you have seen or experienced unacceptable behavior or anything that would make
our community less welcoming, please speak up! See [our code of

> conduct](./code-of-conduct.md) for more information on how to report bad behavior.

Author: I'm using unacceptable for now to unblock this, but I think another adjective might be warranted.

While we want to encourage everyone to express themselves in their own way,
there are some behaviors that we encourage you to adopt while interacting with
other community members.

> #### Act on behalf of the project

Reviewer 3: mega-nitpick, sorry: in the sections above, we used `###` (three levels) for this next level of headers, but here we use `####` (four levels). we should be consistent...

Author: I like mega-nitpick, keep them coming.

It’s no secret that a good number of maintainers of the project are employed by
companies with commercial interests in OpenTelemetry, especially vendors in the
observability space. That said, we expect community members to act in the best
interests of the project. Each member’s priorities can (and should!) align with
those of their employers so that the relationship is beneficial to all parties,
but when acting as a maintainer or contributor to the project, community members
are expected to wear the project’s hat. 

> ##### Disclose potential conflicts of interest

Reviewer 3: do we want to explicitly mention conflicts regarding employers (and/or customers of employers)? the "hats" below are all OTel-specific hats...

Author: Can you give me an example of such conflict? I think I see what you mean, but when taking the previous point into consideration ("wear the project's hat"), I'm not sure what else to cover.

I drafted something like this, but wasn't 100% happy:

> a Collector maintainer might be part of the Governance Committee, a JavaScript maintainer might be part of the Technical Committee, a Go approver might have a customer being affected by a specific behavior on the Go's SDK, ...

Author: I'm marking this as resolved, as we can tweak this later.

during a GC call, a person who is also a maintainer of the Collector might say:
“as a Collector maintainer, I believe that…, while as a GC member, I believe …”

### Assume positive intent

We all have different priorities in our daily jobs, and while some of us are
employed to work full time on OpenTelemetry, some of us are paid to improve
specific parts of the project according to the commercial interests of our
employers. When reviewing proposals, documents, or code, take the different
perspectives into consideration, but more importantly, assume positive intent:
while the proposal might seem skewed towards a specific perspective at first,
it’s very likely that the author is open to improving it if different
perspectives are provided.

### Respectfully disagree

Many decisions are made every day as part of our project. Despite giving our
best, not all decisions are the right ones. We encourage ideas and solutions to
be proposed and debated until an agreement is reached or until the “disagree and
commit” stage is reached. What we cannot tolerate is turning attacks against
ideas into attacks against people: in the heat of the moment, it might be

> tempting to do an ad hominem attack but it’s always wrong. If you have reasons

Reviewer 4: ```suggestion
tempting to make an ad hominem attack but it’s always wrong. If you have reasons
```

> to believe the person you are debating with is not acting on behalf of the

Reviewer 5: ```suggestion
to believe the person you are debating with is not acting in the interest of the
```

project, seek mediation instead of engaging further. While the technical merits
of the matter should be resolved within the SIG by the maintainers or, in
ultimate cases, by the Technical Committee (TC), non-technical matters should be
brought up to the Governance Committee.

### Be nice

As evidenced in [The Cultural Map](https://erinmeyer.com/books/the-culture-map/)
by Erin Meyer, people from different cultural backgrounds have different ways of
communicating. While we don’t expect you to be an expert in [“reading the

> air”](https://www.bbc.com/worklife/article/20200129-what-is-reading-the-air-in-japan),

Reviewer 6: Minor, since the link hinted a particular country which might be considered as bias, should we avoid it here?

Author: Not sure I see it the same way you do, but given @arminru seems to agree with you, I'll reword this.
